modules/tinymce/src/core/main/ts/mode/Mode.ts

```diff
@@ -29,7 +29,7 @@ const switchToMode = (editor: Editor, activeMode: Cell<string>, availableModes:
 };
 
 const setMode = (editor: Editor, availableModes: Record<string, EditorModeApi>, activeMode: Cell<string>, mode: string): void => {
-  if (mode === activeMode.get() || Disabled.isDisabled(editor)) {
+  if (mode === activeMode.get() || (editor.initialized && Disabled.isDisabled(editor))) {
```

Comment:
Previously, the editor could only switch between readonly mode and disabled mode. However, the disabled mode should prevent any mode switching.
The issue is in the init: We should switch to readonly mode before the editor's init phase when both readonly: true and disabled: true are set. Without this check, if we set both readonly: true and disabled: true, the editor incorrectly stays in design mode (while appearing disabled). Then, when we later unset the disabled mode, the editor incorrectly reverts back to design mode instead of staying in readonly mode.https://github.com/tinymce/tinymce/blob/e286406a17d4d16821796fcf883bd443b0008112/modules/tinymce/src/themes/silver/main/ts/UiState.ts#L60

Reply:
I thought that might be the reason 👍 I guess given we don't have an internal bypass and are using the APIs `mode` APIs like an integrator we need the `initialized` check to make some initial changes